lib/_http_outgoing.js

```diff
@@ -138,15 +138,15 @@ OutgoingMessage.prototype._writeRaw = function(data, encoding, callback) {
       this.connection.writable &&
       !this.connection.destroyed) {
     // There might be pending data in the this.output buffer.
-    while (this.output.length) {
-      if (!this.connection.writable) {
-        this._buffer(data, encoding, callback);
-        return false;
+    var outputLength = this.output.length;
+    if (outputLength) {
+      for (var i = 0; i < outputLength; i++) {
+        var cb = this.outputCallbacks[i];
```

Comment:
You mean hoist the `var cb` outside the loop?  It shouldn't matter, V8's flow analysis doesn't care.  The variable as such probably doesn't even exist in the generated machine code.